csrc/python_frontend/fusion_record.h

```diff
@@ -2817,110 +2817,90 @@ struct RandomOpRecord : RecordFunctor {
   PrimDataType dtype_;
 };
 
-//! Specialized Record Functor for recording FusionState vector of sizes for
-//! both inputs and constants.
+//! Specialized Record Functor for recording Vector of Scalars
 
-template <typename ValueType>
 struct VectorRecord : RecordFunctor {
   VectorRecord(
+      std::vector<State> _args,
       std::vector<State> _outputs,
-      serde::RecordType record_type,
-      std::optional<std::vector<ValueType>> value,
-      size_t size,
       PrimDataType dtype)
-      : RecordFunctor({}, std::move(_outputs), "define_vector", record_type),
-        value_(std::move(value)),
-        size_(size),
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "define_vector",
+            serde::RecordType_Vector),
         dtype_(dtype) {}
   ~VectorRecord() override = default;
   RecordFunctor* clone() final {
     return new VectorRecord(*this);
   }
 
   //! Child specific hash function in lower 32 bits.
-  //! | 31 --------------- 16 | 15 ---------------  0 |
-  //! | Dtype                 | Size                  |
+  //! | 31 ---------------------------------------  0 |
+  //! | Dtype                                         |
   size_t hash() const final {
     auto result = RecordFunctor::hash();
-    result |= (static_cast<size_t>(dtype_) & 0xffff) << 16;
-    return result | (size_ & 0xffff);
+    return result | (static_cast<size_t>(dtype_) & 0xffffffff);
```

Comment:
Interesting. so we are basically relying on the inputs feed to the VectorRecord to differentiate them.
Out of curiosity, does the order of these op matter?

i.e. would the two fusion record be considered identical, even though the order of the constant definition is different?
```
const_5 = fd.define_constant(5)
const_6 = fd.define_constant(6)
vector = fd.define_vector([const_5, const_6])
```

```
const_6 = fd.define_constant(6)
const_5 = fd.define_constant(5)
vector = fd.define_vector([const_5, const_6])
```

Reply:
For combining hashes in a way that spreads bits and is order dependent, we may want to update `RecordFunctor::hash` to do something more like `boost::hash_combine`. I previously played with putting something like that into `csrc/utils.h` but I think I didn't merge it.

Reply:
Does combining hash work for this? I thought we would have a cache miss if the order of the definition doesn't match.

Reply:
Why do we need to combine hashes?  The hash matching should be composed based on the use of `define_scalar` as inputs to `define_vector`.  The weakness is it true I deferred figuring out how to hash values for `define_scalar`.  Order does matter as it does with all matching in the caching.